Composer/packages/lib/code-editor/src/LuEditor.tsx

```diff
@@ -91,6 +78,49 @@ const LuEditor: React.FC<LULSPEditorProps> = (props) => {
     editorId = [projectId, fileId, sectionId].join('/');
   }
 
+  const [editor, setEditor] = useState<any>();
+
+  useEffect(() => {
+    if (!editor) return;
+
+    if (!window.monacoServiceInstance) {
+      window.monacoServiceInstance = MonacoServices.install(editor as any);
+    }
+
+    const uri = get(editor.getModel(), 'uri._formatted', '');
+    const url = createUrl(luServer);
+    const webSocket: WebSocket = createWebSocket(url);
+    listen({
+      webSocket,
+      onConnection: (connection: MessageConnection) => {
+        const languageClient = createLanguageClient('LU Language Client', ['lu'], connection);
+        if (!window.monacoLUEditorInstance) {
+          window.monacoLUEditorInstance = languageClient;
+        }
+
+        const m = monacoRef.current;
+        if (m) {
+          editor.addCommand(m.KeyMod.Shift | m.KeyCode.Enter, function () {
+            const position = editor.getPosition();
+            SendRequestWithRetry(languageClient, 'labelingExperienceRequest', { uri, position });
+          });
+        }
+
+        SendRequestWithRetry(languageClient, 'initializeDocuments', { luOption, uri });
```

Comment:
Shouldn't this be awaited to ensure the document has been initialized?

Reply:
@zhixzhan I am going to go ahead and take this but please respond here when you get back online and create a follow-up PR if this needs to be fixed.

Reply:
I checked code below `SendRequestWithRetry(languageClient, 'initializeDocuments', {`, nothing relied on it.
One effect is before `initializeDocuments `, user interact in LSP editor interface, all information needs come from LSP server ( diagnostics/suggustions...) would be empty, looks like writing in a plain text-editor.